securitycenter/bigqueryv2/bigquery_test.go

```diff
@@ -0,0 +1,287 @@
+	}
+
+	if err := client.DeleteBigQueryExport(ctx, req); err != nil {
+		return fmt.Errorf("failed to delete BigQueryExport: %w", err)
+	}
+
+	return nil
+}
+
+func TestListBigQuery(t *testing.T) {
+
+	testutil.Retry(t, 5, 20*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+
+		// Create Test BigQueryExport Config
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			t.Fatalf("Issue generating id.")
+			return
+		}
+		configID := "random-bqexport-id-" + rand.String()
+
+		if err := addBigQueryExport(t, configID); err != nil {
+			t.Fatalf("Could not setup test environment: %v", err)
+			return
+		}
+
+		parent := fmt.Sprintf("organizations/%s/locations/global", orgID(t))
+
+		// Call List BigQueryExport
+		err = listBigQueryExport(buf, parent)
+
+		if err != nil {
+			t.Fatalf("listBigQueryExport() had error: %v", err)
+			return
+		}
+
+		got := buf.String()
+
+		if !strings.Contains(got, configID) {
+			t.Fatalf("listBigQueryConfigs() got: %s want %s", got, configID)
+		}
+
+		// Cleanup
+		cleanupBigQueryExport(t, configID)
+	})
+}
+
+func TestGetBigQuery(t *testing.T) {
+
+	testutil.Retry(t, 5, 20*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		// Create Test BigQueryExport Config
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			t.Fatalf("Issue generating id.")
+			return
+		}
+		configID := "random-bqexport-id-" + rand.String()
+
+		if err := addBigQueryExport(t, configID); err != nil {
+			t.Fatalf("Could not setup test environment: %v", err)
+			return
+		}
+
+		parent := fmt.Sprintf("organizations/%s/locations/global", orgID(t))
+
+		// Call GetBigQueryExport
+		err = getBigQueryExport(buf, parent, configID)
+
+		if err != nil {
+			t.Fatalf("getBigQueryExport() had error: %v", err)
+			return
+		}
+
+		got := buf.String()
+
+		if !strings.Contains(got, configID) {
+			t.Fatalf("getBigQueryExport() got: %s want %s", got, configID)
+		}
+
+		// Cleanup
+		cleanupBigQueryExport(t, configID)
+	})
+}
+
+func TestDeleteBigQuery(t *testing.T) {
+
+	testutil.Retry(t, 5, 20*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		// Create Test BigQueryExport Config
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			t.Fatalf("Issue generating id.")
+			return
+		}
+		configID := "random-bqexport-id-" + rand.String()
+
+		if err := addBigQueryExport(t, configID); err != nil {
+			t.Fatalf("Could not setup test environment: %v", err)
+			return
+		}
+
+		parent := fmt.Sprintf("organizations/%s/locations/global", orgID(t))
+
+		// Call DeleteBigQueryExport
+		err = deleteBigQueryExport(buf, parent, configID)
+
+		if err != nil {
+			t.Fatalf("deleteBigQueryExport() had error: %v", err)
+			return
+		}
+
+		got := buf.String()
+
+		if !strings.Contains(got, configID) {
+			t.Fatalf("deleteBigQueryExport() got: %s want %s", got, configID)
+		}
+	})
+}
+
+func TestUpdateBigQuery(t *testing.T) {
+
+	testutil.Retry(t, 5, 20*time.Second, func(r *testutil.R) {
```

Comment:
Same as previous: consider removing this from the `testutil.Retry()` method. That said, I could understand if the update operation tends to be a bit more flaky--if so, reduce the number of retries to 3 and the time to 10 seconds.

Reply:
Addressed